src/codeowners/mod.rs

```diff
@@ -0,0 +1,193 @@
+mod audit;
+/// Use an iterative approach, i.e., scan repos one by one
+mod iterate;
+/// Leverage the GH search API to find relevant information
+mod search;
+
+use std::{collections::HashSet, fmt::Display};
+
+use crate::{members::get_indexed_org_members, teams::get_indexed_org_teams, Bootstrap};
+use colored::Colorize;
+use lazy_static::lazy_static;
+use regex::Regex;
+
+lazy_static! {
+    /// Regex used to find matches in CO files that look like @something.
+    static ref co_regex: Regex = Regex::new(r"@\S+").unwrap();
+}
+
+/// Represents a CODEOWNERS file
+#[derive(Debug)]
+struct CodeownersFile {
+    /// Repository this CO file belongs to
+    repo: String,
+    /// HTML URL, to give the user a quick way to reach the file
+    url: String,
+    /// List of all users mentioned in the file, for further analysis
+    users: HashSet<String>,
+    /// List of all teams mentioned in the file, for further analysis
+    teams: HashSet<String>,
+}
+
+impl CodeownersFile {
+    /// Process the content of a CO file and turn it into a CodeownersFile struct
+    fn parse_from_content(
+        bootstrap: &Bootstrap,
+        content: String,
+        html_url: &str,
+        repo: &str,
+    ) -> CodeownersFile {
+        let mut users = HashSet::new();
+        let mut teams = HashSet::new();
+
+        // The prefix that teams have in CO files
+        let team_prefix = format!("@{}/", bootstrap.org);
+
+        for line in content.split('\n') {
+            if line.trim().starts_with('#') {
+                // Skip comments
+                continue;
+            }
+
+            for m in co_regex.find_iter(line) {
+                let matched = m.as_str();
+                if matched.starts_with(&team_prefix) {
+                    // It's a team
+                    teams.insert(matched.trim_start_matches(&team_prefix).to_string());
+                } else {
+                    // It's a user
+                    users.insert(matched.trim_start_matches("@").to_string());
+                }
+            }
+        }
+
+        CodeownersFile {
+            repo: repo.to_string(),
+            url: html_url.to_string(),
+            users,
+            teams,
+        }
+    }
+}
+
+/// The errors we can encounter while auditing a CODEOWNERS file
+enum CodeownersFileError {
```

Comment:
Unify the error and warning types and would add another enum to wrap the API returned ones